iothub_client/src/iothubtransport_amqp_common.c

```diff
@@ -775,8 +791,17 @@ static void on_amqp_connection_state_changed(const void* context, AMQP_CONNECTIO
         // Codes_SRS_IOTHUBTRANSPORT_AMQP_COMMON_09_060: [If `new_state` is AMQP_CONNECTION_STATE_ERROR, the connection shall be flagged as faulty (so the connection retry logic can be triggered)]
         if (new_state == AMQP_CONNECTION_STATE_ERROR)
         {
+            MUX_CONN_STATUS_CB_ARGS args;
+            args.status = IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED; 
+            args.reason = IOTHUB_CLIENT_CONNECTION_NO_NETWORK;
+
             LogError("Transport received an ERROR from the amqp_connection (state changed %s -> %s); it will be flagged for connection retry.", MU_ENUM_TO_STRING(AMQP_CONNECTION_STATE, previous_state), MU_ENUM_TO_STRING(AMQP_CONNECTION_STATE, new_state));
-            transport_instance->transport_callbacks.connection_status_cb(IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED, IOTHUB_CLIENT_CONNECTION_NO_NETWORK, transport_instance->transport_ctx);
+            
+            (void)singlylinkedlist_foreach(
```

Comment:
>singlylinkedlist_foreach [](start = 18, length = 24)

I'm not that familiar with this code, but... why are we only signaling this on errors?  Shouldn't caller get a callback on the network being available also?

Reply:
Hey John, it does get invoked for connection re-established. Please look here: https://github.com/Azure/azure-iot-sdk-c/blob/2020-01-22/iothub_client/src/iothubtransport_amqp_common.c#L293

Reply:
OK - thanks for followup and offline conversation.  Flipping to approved.